save-cloud-common/src/commonMain/kotlin/org/cqfn/save/domain/Sdk.kt

```diff
@@ -18,25 +18,52 @@ open class Sdk(val name: String, open val version: String) {
     object Default : Sdk("ubuntu", "latest")
 
     override fun toString() = "$name:$version"
+
+    companion object {
+        val sdks = listOf("Default", "Java", "Python")
+    }
 }
 
 /**
  * @property version version of JDK
  */
-class Jdk(override val version: String) : Sdk("openjdk", version)
+class Jdk(override val version: String) : Sdk("openjdk", version) {
+    companion object {
+        val versions = listOf("8", "11")
+    }
+}
 
 /**
  * @property version version of Python
  */
-class Python(override val version: String) : Sdk("python", version)
+class Python(override val version: String) : Sdk("python", version) {
+    companion object {
+        val versions = listOf("2.7", "3.7")
+    }
+}
 
 /**
  * Parse string to sdk
+ *
+ * @return sdk by string
+ */
+fun String.toSdk(): Sdk {
+    val splitSdk = this.split("$")
+    require(splitSdk.size == 2) { "Cant find correct sdk and version" }
+    val (sdkType, sdkVersion) = splitSdk.run { this.first() to this.last() }
+    return when (sdkType) {
+        "Java" -> Jdk(sdkVersion)
```

Comment:
This is still a pretty bad hard-code. You only need to choose it in human-readable form on frontend, from there you can encode it into json and let serialization library do all the work

Reply:
You could move list of sdk names outside of `Sdk` and reference class fields there, e.g. `val sdks = listOf(Jdk.name, Python.name)`, and also reference same properties in this `when`

Reply:
Now it has already been implemented

Reply:
Where?

Reply:
My answer was for first comment) What do you mean `and also reference same properties in this when`?

Reply:
```kotlin
data class Jdk(val version) {
    companion object { val name = "JDK" }
}

fun String.toSdk() = when (this) {
    Jdk.name - >Jdk(version)
}
```
like this, now you are repeating `"Java"` literal multiple times

Reply:
Fixed